doc/manual/src/language/string-context.md

```diff
@@ -0,0 +1,145 @@
+> "Regular" strings an empty context are the only ones inspected, e.g. compared with `==`.
+
+String context elements come in 3 forms:
+
+- [*constant*]{#string-context-element-constant}
+- [*output*]{#string-context-element-output}
+- [*derivation deep*]{#string-context-element-derivation-deep}
+
+*Constant* and *output* string contexts elements are just
+[deriving paths](@docroot@/glossary.md#gloss-deriving-path);
+those are just the names of the two kinds of deriving path.
+See the documentation on deriving paths for further details.
+
+*derivation deep* is an advanced feature intended to be used with the
+[`exportReferencesGraph`](./advanced-attributes.html#adv-attr-exportReferencesGraph)
+advanced derivation feature.
+A *derivation deep* string context element is a derivation path, and refers to both its outputs and the entire build closure of that derivation:
+all its outputs, all the other derivations the given derivation depends on, and all their outputs too.
+
+## Inspecting string contexts
+
+Most basically, [`builtins.hasContext`] will tell whether a string has a non-empty context.
+
+When more granular information is needed, [`builtins.getContext`] can be used.
+It creates an [attribute set] representing the string context; the attribute set can be inspected normally.
+
+[`builtins.hasContext`]: ./builtins.md#builtins-hasContext
+[`builtins.getContext`]: ./builtins.md#builtins-getContext
+[attribute set]: ./values.md#attribute-set
+
+## Clearing string contexts
+
+[`buitins.unsafeDiscardStringContext`](./builtins.md#) will make a copy of a string, but with an empty string context.
+The returned string can be used in more ways, e.g. by operators that require that the string context is empty.
+The requirement to explicitly discard the string context in such use cases, helps ensure that string context elements are not lost by mistake.
+The "unsafe" marker is only there to remind that Nix normally guarantees that dependencies are tracked, whereas the returned string has lost them.
+
+## Constructing string contexts
+
+[`builtins.appendContext`] will create copy of a string but with additional string context elements.
+The context is specified explicitly by an [attribute set] in the format that [`builtins.hasContext`] produces.
+We can create strings with arbitrary contexts in 3 steps:
+
+1. Create strings with the string context elements we want
+2. Dump their contexts with [`builtins.getContext`]
+3. Combine them together with a base string and repeated [`builtins.appendContext`] calls.
+
+The remainder of this section will focus on step 1: making strings with individual string context elements on which to apply `builtins.getContext`.
+
+[`builtins.appendContext`]: ./builtins.md#builtins-appendContext
+
+## Constant string context elements
+
+A constant string context element is just a constant [deriving path];
+a constant deriving path is just a [store path].
```

Comment:
I am trying to distinguish the proverbial "sum type data constructor" vs the single field.

"constant deriving path" in *colloquial* speech is just a synonymy for "store path", but it would be mean to create redundant jargon words for the same constant. Formally "constant" and "output", describe the two cases of "deriving path". I guess we can think of these data structures as adjectives:

>  A deriving path is either "constant" [adj] in which case it is a store path, or "output" (forced adjective, awk) in which case it a pair of a store path an an output name.

Reply:
My using "just" was the most abbreviated and unclear (and subconcious) way of dealing

Reply:
Does the suggestion still make sense then?